test/test_blocks.py

```diff
@@ -465,6 +469,57 @@ def test_raise_error_if_event_queued_but_queue_not_enabled(self):
         demo.close()
 
 
+class TestTempFile:
+    def test_pil_images_hashed(self):
+        def create_images(n_images):
+
+            a = Image.new("RGB", (512, 512), "red")
+            b = Image.new("RGB", (512, 512), "green")
+            c = Image.new("RGB", (512, 512), "blue")
+
+            res = [a, b, c][:n_images]
+            random.shuffle(res)
+            return res
+
+        dir_ = tempfile.mkdtemp()
+
+        try:
+            with mock.patch.dict(os.environ, {"GRADIO_TEMP_DIR": dir_}):
+                demo = gr.Interface(
+                    create_images,
+                    inputs=[gr.Slider(value=3, minimum=1, maximum=3, step=1)],
+                    outputs=[gr.Gallery().style(grid=2, preview=True)],
+                )
+                _, url, _ = demo.launch(prevent_thread_lock=True)
+                client = grc.Client(url)
+                _ = client.predict(3)
+                _ = client.predict(3)
+            # only three files created
+            assert len(list(pathlib.Path(dir_).glob("**/*"))) == 3
+        finally:
+            demo.close()
+            shutil.rmtree(dir_)
+
+    def test_no_empty_files(self):
+        file_dir = pathlib.Path(pathlib.Path(__file__).parent, "test_files")
+        video = str(file_dir / "video_sample.mp4")
+        dir_ = tempfile.mkdtemp()
```

Comment:
You can use the `tmp_path` Py.test fixture to avoid doing this by hand.

Reply:
Done! Thanks for the suggestion